### DISCUSSION TITLE
Run quarkusIntTest against a running application using gradle

### DISCUSSION THREAD

**Discussion by guitcastro** [quarkusio/quarkus#27491] [Q&A]
The [testing guide](https://quarkus.io/guides/getting-started-testing#executing-against-a-running-application) states that you can run integration tests using against a running app using: 

`./mvnw verify -Dquarkus.http.test-host=1.2.3.4 -Dquarkus.http.test-port=4321`

I am trying to the the same thing, but using gradle:

`./gradlew quarkusIntTest -Dquarkus.http.test-host=1.2.3.4 -Dquarkus.http.test-port=4321`

However it did not worked.

[a comment by an automated account is not shown]

**Comment by kaja78** [comment]
I am facing the same issue. 

For `testNative` and `quarkusIntTest` gradle tasks, I am getting:
`Unrecognized configuration key "quarkus.http.test-host" was provided; it will be ignored; verify that the dependency extension for this configuration is set or that you did not make a typo`

Any updates on this?

**Reply by glefloch** [reply]
Not yet, I need to look into it.

**Comment by kaja78** [comment]
As a workaround I am using custom [JVM Test Suite](https://docs.gradle.org/current/userguide/jvm_test_suite_plugin.html) (testApp) to run my non-quarkus integration tests against running app using `gradle testApp`:
```
testing {
    suites {
        val test by getting(JvmTestSuite::class) {
            useJUnitJupiter()
        }

        val testApp by registering(JvmTestSuite::class) {
            dependencies {
                implementation(project())
                implementation("io.rest-assured:rest-assured:4.5.1")
            }
            targets {
                all {
                    testTask.configure {
                        shouldRunAfter(test)
                    }
                }
            }
        }
    }
}
```

**Comment by bonneval** [comment]
I`m facing the same issue.

When running
`./gradlew quarkusIntTest -Dquarkus.http.test-host=1.2.3.4 -Dquarkus.http.test-port=4321`

Quarkus starts the jar file and ignores the provided properties.
`Executing "/usr/lib/jvm/java-11-openjdk-amd64/bin/java -Dquarkus.http.port=8081 -Dquarkus.http.ssl-port=8444 -Dtest.url=http://localhost:8081 ...`

The cause of the problem is that following line
https://github.com/quarkusio/quarkus/blob/5611919e07deb189c5d414c36a91a432ce9431b2/test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusIntegrationTestExtension.java#L257
does not retrieve the property which is set via `gradle -D...` or via any property file.

I found a workaround using QuarkusTestProfile:
```
public class IntegrationTestProfile implements QuarkusTestProfile {
    @Override
    public Map<String, String> getConfigOverrides() {
        return Map.of(
                "quarkus.http.test-host", "1.2.3.4",
                "quarkus.http.test-port", "4321"
        );
    }
}
```
Unfortunately this is not as flexible as I like, but that is a starting point.
Maybe some of the Devs can explain why we face this behavior and hopefully use this as a hint for a BugFix.

**Reply by boekkooi-lengoo** [reply]
@glefloch I'm really sorry to bug but this still seems to be an issue in 3.6.6. Do you maybe have some time to look at this? 
Thanks you in advance.
